### PR TITLE
[WFCORE-4842] Add tests for TLS 1.3 that make use of the OpenSSL TLS provider

### PR DESCRIPTION
**Pull request by fjuma**
https://issues.redhat.com/browse/WFCORE-4842
https://issues.redhat.com/browse/EAP7-1414

To pass CI, this will require a WildFly OpenSSL upgrade but this is ready to be reviewed.

Analysis document: https://github.com/wildfly/wildfly-proposals/pull/288
WildFly PR with documentation: https://github.com/wildfly/wildfly/pull/13414
Depends on: https://github.com/wildfly-security/wildfly-openssl/pull/81

### CODE REVIEW

**Comment by darranl** [comment]
This PR is awaiting the RFE process to be merged.

**Review by jstourac — APPROVED** [review]

**Comment by fjuma** [comment]
Just FYI, this one has been pre-checked and depends on https://github.com/wildfly/wildfly-core/pull/4329.

File: elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java [hunk 1]
```diff
@@ -345,16 +362,24 @@ public void testSslServiceAuth() throws Throwable {
...
+        testCommunication("ServerSslContextTLS13OpenSsl", "ClientSslContextTLS13OpenSsl", false, "OU=Elytron,O=Elytron,C=CZ,ST=Elytron,CN=localhost",
+                "OU=Elytron,O=Elytron,C=UK,ST=Elytron,CN=Firefly", "TLS_AES_256_GCM_SHA384", true);
+    }
+
     @Test
     public void testSslServiceAuthProtocolMismatch() throws Throwable {
-        Assume.assumeTrue("Skipping testSslServiceAuthProtocolMismatch, test is not being run on JDK 11.",
-                System.getProperty("java.specification.version").equals("11"));
+        Assume.assumeTrue("Skipping testSslServiceAuthProtocolMismatch, test is not being run on JDK 11+.",
+                JdkUtils.getJavaSpecVersion() >= 11);
```
Comment (jstourac) [inline elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java:392]: @fjuma, just noticed - isn't here missing the `&& isOpenSSL111OrHigher()` part here? There is client using TLS13 two lines down...

**Review by darranl — APPROVED** [review]

**Comment by darranl** [comment]
FYI I have removed the feature label as at this point in time this is now just test cases for an already merged feature.

**Pull request merged**